backend/app/models/content.py

```
@@ -0,0 +1,118 @@
+    """Base model for content items, containing common fields."""
+    user_id: uuid.UUID = Field(foreign_key="user.id", index=True)
+    type: str = Field(
+        sa_column_args=[CheckConstraint("type IN ('url', 'pdf', 'docx', 'text', 'plugin')")],
+        max_length=50,
+        index=True,
+    )
+    source_uri: Optional[str] = Field(default=None, max_length=2048)
+    title: Optional[str] = Field(default=None, max_length=255)
+    summary: Optional[str] = Field(default=None)
+    content_text: Optional[str] = Field(default=None)
+    content_vector: Optional[List[float]] = Field(default=None, sa_column=Column(JSONB))
```

Ruff (0.11.9)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)